da/celestia/celestia.go

```diff
@@ -629,3 +604,46 @@ func (c *DataAvailabilityLayerClient) getDataAvailabilityHeaders(height uint64)
 
 	return headers.DAH, nil
 }
+
+// Celestia syncing in background
+func (c *DataAvailabilityLayerClient) syncCelestia(rpc *openrpc.Client) {
+	sync := func() error {
+		done := make(chan error, 1)
+		go func() {
+			done <- rpc.Header.SyncWait(c.ctx)
+		}()
+		ticker := time.NewTicker(30 * time.Second)
+		defer ticker.Stop()
+		for {
+			select {
+			case err := <-done:
+				if err != nil {
+					c.logger.Error("Failed to sync Celestia DA.", "Error", err)
+				}
+				return err
+			case <-ticker.C:
+				state, err := rpc.Header.SyncState(c.ctx)
+				if err != nil {
+					c.logger.Error("Failed to sync Celestia DA.", "Error", err)
+					return err
+				}
+				c.logger.Info("Celestia-node syncing", "height", state.Height, "target", state.ToHeight)
+
+			}
+		}
+	}
+
+	err := retry.Do(sync,
```

Comment:
Instead of logging the error in the function I think we can just return a wrapped error and change the retry.Do to something like the following:

```
err := retry.Do(sync,
		retry.Attempts(0), // try forever
		retry.Delay(10*time.Second),
		retry.LastErrorOnly(true),
		retry.DelayType(retry.FixedDelay),
		retry.OnRetry(func(n uint, err error) {
			c.logger.Error("Retry sync due to error", "attempt", n, "error", err)
		}),
	)
```

Reply:
done